Develop

meridian285 opened the pull request:
Добавлены 4 теста: 
Сравнение веса двух покемонов
Проверка на наличие способности у покемона 'run-away'
Проверка что у каждого покемона в списке есть name не пустое, запрос выдает заданное число покемонов
Проверка что заданный лимит списка выдает нужное колличество покемонов

используемые библиотеки:
allure-maven
junit-jupiter-engine
json
rest-assured
allure-junit5
allure-rest-assured

meridian285 committed:
Добавить начальная стуктура проекта

Config.java класс с URL проекта
Pokemon.java класс десереализации, abilities, weight
RestClient спецификация
pom добавлены библиотеки junit5, rest-assured, apache.commons, jackson, google.code.gson,

meridian285 committed:
Добавить test getWeight

meridian285 committed:
Добавить test getWeight

meridian285 committed:
Добавить эксперементы)

meridian285 committed:
Добавить изменения в checkAbility() осталось разобраться с проверкой

checkWeightPokemon() все еще не работает через Pokemon.class

meridian285 committed:
Добавить готов тест на проверку наличия способности checkAbility()

meridian285 committed:
Добавить изменения checkWeightPokemon() тест закончен и работает

meridian285 committed:
Добавить тестовый класс checkNamePokemon()

Тест проверки поля name в списке покемонов
Добавлены комментарии во все классы

meridian285 committed:
Добавить класс Result для pojo

meridian285 committed:
Добавить тестовый метод checkLimitPokemonList(), для проверки колличества покемонов

meridian285 committed:
Добавить проверку в тест checkListPokemon()

заполнен README.md

meridian285 committed:
Добавить

meridian285 committed:
Добавить изменения в pom, и .filter(new AllureRestAssured()) в getWeightPokemon

meridian285 committed:
Добавить в отчет request, response тестов.

Несколько переименований

sun6r0 commented on pom.xml:
```diff
@@ -9,8 +9,160 @@
+<!--            <groupId>org.asynchttpclient</groupId>-->
+<!--            <artifactId>async-http-client</artifactId>-->
+<!--            <version>2.12.3</version>-->
+<!--        </dependency>-->
```
Неиспользуемые зависимости и прочую грязь перед мержами и пушами убираем всегда

sun6r0 commented on pom.xml:
```diff
@@ -9,8 +9,160 @@
+<!--            </plugin>-->
+<!--        </plugins>-->
+<!--    </build>-->
+<!--</project>-->
```
То же самое

sun6r0 commented on src/test/java/co/pokeapi/steps/PokemonSteps.java:
```diff
@@ -0,0 +1,54 @@
+    public int getWeightPokemon(String namePokemon) {
+        return given()
+                .when()
+                .filter(new AllureRestAssured())
```
Фильтр лучше в спецификацию вынести

sun6r0 commented on src/test/java/co/pokeapi/tests/PokemonStatsTests.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    @Test
+    @DisplayName("Проверка что у каждого покемона в списке есть name не пустое, запрос выдает заданное число покемонов")
```
Название теста выглядит так, будто он проверяет, пришло ли заданное число покемонов, но он этого не делает, лучше назвать "Проверка, что при запросе 1000 покемонов, у каждого покемона в списке не пустое name"

meridian285 committed:
Добавить комментарий в тестовый класс PokemonTests

Добавить test_suite.XLSX тестовый набор

meridian285 committed:
Добавить фильтр .filter(new AllureRestAssured()) вынесен в спецификацию, исправленно название одного теста на более корректное

meridian285 committed:
Добавить переделаны шаги для тестов, и сами тесты

В шагах удалены методы getAbility и getWeightPokemon, вместо них теперь один метод getSameFields
Тесты соответственно тоже переделаны
В pom добавлен Ломбок
Удалены все pojo классы кроме Root, остальные классы теперь в нем
В классе EndPoints.java добавлены более универсальные эндпоинты
